### ISSUE TITLE
Set the default keyring backend to "test"

### ISSUE THREAD

**Issue by evan-forbes** [celestiaorg/celestia-app#1201]
Per a synchronous decision, the extra security provided by using an encrypted file on disk for the key is just that, extra. For most usecases, using a burner key will be fine, and this dramatically simplifies the implementation efforts. Also, if more security is needed, then a key protected by hardware will most likely be used.

**Comment by rootulp** [comment]
See https://github.com/celestiaorg/celestia-node/pull/1565#issuecomment-1372199529 for details.

**Comment by rootulp** [comment]
Related https://github.com/celestiaorg/celestia-app/pull/1141

**Comment by rootulp** [comment]
Based on my investigation (using [debug.go](https://github.com/rootulp/cosmos-sdk/blob/rp/keys-debug/client/keys/debug.go)) I don't think the method of overriding the default keyring backend inside celestia-app (see [here](https://github.com/rahulghangas/celestia-app/blob/7bb4bba5688e913847f03752aa98e2bf7dd622c6/cmd/celestia-appd/cmd/root.go#L154-L156)) does what we expect. With this approach I observe:

```shell
celestia-appd init foo
cat ~/.celestia-app/confg/client.toml

// trimmed output
keyring-backend = "os"
``` 

| client.toml | `--keyring-backend` | keyring backend in use |
| --- | --- | --- |
| os   | unspecified  | os  |
| os   | file  | file  |
| file   | unspecified  | file  |
| file   | os  | os  |

In other words, the value specified by the CLI flag takes precedence over the value in client.toml
But the default value of `file` never gets written to `client.toml` so the help message is incorrect:

```
      --keyring-backend string   Select keyring's backend (os|file|test) (default "file")
```
